cluster/addons/addon-manager/kube-addons.sh

```diff
@@ -193,6 +194,10 @@ for obj in $(find /etc/kubernetes/admission-controls \( -name \*.yaml -o -name \
   log INFO "++ obj ${obj} is created ++"
 done
 
+# Create Istio resources, which are in istio-system namespace.
+log INFO "== Applying Istio addon (if present) at $(date -Is) =="
+${KUBECTL} ${KUBECTL_OPTS} apply -f ${ISTIO_PATH} --recursive | grep -v configured
```

Comment:
Why do we need this extra step? Are Istio addon manifests not under ${ADDON_PATH}?

Reply:
They are there but it's a long manifest and not all of the resources are labeled with CLUSTER_SERVICE_LABEL, only the deployments at the moment so the apply fails for some resources unless they are already created.

Reply:
Thanks for the clarification, it sounds like we need a way to express the creation order. I wonder if there is a more generic way to achieve this.

Note that the resources without any of `CLUSTER_SERVICE_LABEL` or `ADDON_MANAGER_LABEL` will not be deleted when its manifest is removed from the folder (e.g. when disabling istio). Addon-manager's cleanup logic is accomplished via `kubectl apply -f {FOLDER} --prune=true -l {LABEL}`, which relies on label matching.

Also, I vaguely remember that current implementation of `kubectl apply --prune` doesn't handle the multi-namespaces cases very well. Let's say we have two manifests, one is under namespace1, the other under namespace2. `kubectl apply -f --prune` works well for cleaning up resource as long as there are at least one resource in each namespace defined in manifests. Once the last resource in that namespace gets remove from the manifest (e.g. again, disabling istio), this resource will not be cleaned up because `kubectl` will not even look into that namespace if it is neither given via `--namespace` flag nor defined in the manifests.

I start feeling that the solution would be messy given the inflexibility of addon-manager...